src/python/pants/engine/subsystem/native.py

```diff
@@ -273,12 +272,24 @@ def extern_project(context_handle, val, field, type_id):
   return c.to_value(projected)
 
 
-@_FFI.callback("ValueBuffer(ExternContext*, Value*, Field*)")
-def extern_project_multi(context_handle, val, field):
+@_FFI.callback("Value(ExternContext*, Value*, uint8_t*, uint64_t)")
+def extern_project_no_type(context_handle, val, field_str_ptr, field_str_len):
+  """Given a Value for `obj`, a field name, and a type, project the field as a new Value."""
```

Comment:
Thought I'd removed this. I'd extracted this patch from some work in progress. I'll rm it as it's unused.